[FEAT] Добавлен параметр ibsrv для запуска команд с использование автономного сервера

yukon39 opened the pull request:
Добавлен параметр ibsrv для запуска команд с использование автономного сервера для команд:
* run
* vanessa
* xunit
* designer

Closes #619 

Требует изменений в библиотеке ibcmdrunner [PR #3](https://github.com/vanessa-opensource/ibcmdrunner/pull/3)

yukon39 committed:
Добавлен параметр --ibsrv для команды run

yukon39 committed:
Добавлен параметр --ibsrv для команды run

yukon39 committed:
Добавлен параметр --ibsrv для команды vanessa

yukon39 committed:
Добавлен параметр --ibsrv для команды vanessa

yukon39 committed:
Добавлен параметр --ibsrv для команды xunit

yukon39 committed:
Добавлен параметр --ibsrv для команды xunit

yukon39 committed:
Добавлен параметр --ibsrv для команды designer

yukon39 committed:
Добавлен параметр --ibsrv для команды designer

yukon39 committed:
Исправление опечатки

yukon39 committed:
Исправление опечатки

yukon39 committed:
Улучшена диагностика ошибок при запуске в режиме предприятия

yukon39 committed:
Улучшена диагностика ошибок при запуске в режиме предприятия

yukon39 committed:
Добавлен подбор портов для запуска ibsrv

yukon39 committed:
Добавлен подбор портов для запуска ibsrv

yukon39 committed:
Установлен рабочий каталог при подготовке контекстов тестирования

yukon39 committed:
Установлен рабочий каталог при подготовке контекстов тестирования

yukon39 committed:
Добавлена отсуствующая зависимость

yukon39 committed:
Добавлено формирование и сипользование файла конфигурации АС

yukon39 committed:
Добавлено формирование и сипользование файла конфигурации АС

yukon39 committed:
Добавлено описание к методу ВыполнитьДействие

yukon39 committed:
Добавлено описание к методу ВыполнитьДействие

Commit:
исправление ошибки запуска на линукс-машинах

Commit:
версия 2.2.22

artbear commented on src/Классы/КомандаЗапуститьВРежимеПредприятия.os:
```diff
@@ -55,20 +50,28 @@
-// Параметры:
-//   ПараметрыКоманды - Соответствие - Соответствие ключей командной строки и их значений
-//   ДополнительныеПараметры - Соответствие - дополнительные параметры (необязательно)
-//
```
зря удалены описания экспортных методов. требуется их восстановить по всем измененным классам

yukon39 replied:
Готово

yukon39 committed:
Возвращены описания обработчиков событый

yukon39 committed:
Возвращены описания обработчиков событый

yukon39 committed:
Улучшена диагностика выполнения команд

yukon39 committed:
Улучшена диагностика выполнения команд

atyz2013 committed:
Добавлен параметр --lockend для указания времени завершения блокировки сеансов (#631)

* Update КомандаУправлениеСеансами.os

добавлен параметр заполнения окончания блокировки сеансов

* Update КомандаУправлениеСеансами.os

если выполняется команда session unlock очищаем конец завершения сеансов

* Update КомандаУправлениеСеансами.os

добавлена проверка что это "Блокировка" для заполнения времени начала блокировки

* Update КомандаУправлениеСеансами.os

atyz2013 committed:
Update МенеджерКонфигуратора.os

добавлена проверка на ожидание завершения при открытии конфигуратора
если открытие конфигуратора происходит интерактивно и не требует ожидания - сообщение "Работа Конфигуратора завершена" не выводится

Segate-ekb committed:
Тест нового ci

Segate-ekb committed:
Merge branch 'develop' of https://github.com/vanessa-opensource/vanessa-runner into feature/new_ci

Segate-ekb committed:
Revert "Тест нового ci"

This reverts commit 55ec9eaedd14a2fbff38aaf61516bd7981a9b303.

Segate-ekb committed:
Переезд ci на Github actions (#635)

* Имплементация нового ci

* Добавление прогона тестов на бубунте

* сменил имя секрета

Segate-ekb committed:
Update testing.yml

Segate-ekb committed:
Update qa.yml

Segate-ekb committed:
Update qa.yml

Segate-ekb committed:
Теперь падение пайпа ожидается на этапе публикации репорта.
Искусственное падение в test.os я убрал

Segate-ekb committed:
Тут будут собираться фиксы тестов для linux (#636)

* Фикс первых тестов

* Добавил в CI прогон юнитов

Segate-ekb committed:
фикс слешей для linux

Segate-ekb committed:
Исправление поведения команды init-dev с флагом --dt (#625)

fix #624

Исправлено поведение команды при загрузке dt.
Убрано лишнее обновление, для случая, если после инициализации бд или загрузки dt больше действий не производилось.
Вернул поведение при котором сначала грузилась база из dt, а потом на нее накатывались исходники. Можно опять использовать конструкции вида init-dev --dt ./xxx.dt --src ./src/cf При этом будет сначала загружен дамп, а потом будут загружены исходники.(cf, src, хранилище)
fix #624

Segate-ekb committed:
Исправлен прогон тестов на linux и windows, запуск сонара в ПР.

Segate-ekb committed:
Объединил бдд и юниты в один файл для корректного рассчета покрытия

Segate-ekb committed:
Merge pull request #638 from Segate-ekb:feature/coverege

Объединил бдд и юниты в один файл для корректного рассчета покрытия

Segate-ekb committed:
Update testing.yml

Segate-ekb committed:
Сменил формат отчета

Segate-ekb committed:
Добавил локаль в винду, без нее ibcmd плохо

Segate-ekb committed:
Обновил локаль

Segate-ekb committed:
Вернул статистику для расчета покрытия

Segate-ekb commented:
@yukon39 Привет, обновились тесты и тд.
Требуется ребейзнуть ПР из девелопа

artbear committed:
фича Приоритеты настроек

и новые переменные окружения
- RUNNER_SRC аналог --src
- RUNNER_NEW_VERSION для --new-version

Segate-ekb committed:
Дополнил тесты
Переименовал тест "Переопределение настроек"

Segate-ekb committed:
Добавил конкаренси

Segate-ekb committed:
Добавил конкаренси

artbear committed:
вернул пропуск. сценарии и пропускаю одно падение

artbear committed:
исправлены ошибки в примерах

closed Ошибка в примере example.env.json #608

artbear committed:
уточнен тест интерактивного запуска конфигуратора

связано с #632

Segate-ekb committed:
Рефакторинг работы с настройками (#643)

* Отдельный ПР с настройкой параметров

* merge settings-test

* Добавил тесты на булево

* Перенес работу с соответствием параметров из общего модуля в приватную часть менеджера команд приложения.

Раскомментировал фича-файл с проверкой настроек.

* фиксы мелких замечаний

* Отдельный ПР с настройкой параметров

* rebase

* фиксы мелких замечаний

* фиксы замечаний

* Тест на подстановку переменных окружения в настройках

* мелкие синтакс-фиксы перед мержем

---------

Co-authored-by: Artur Ayukhanov <aartbear@gmail.com>

artbear committed:
убрал отладочный вывод в фиче

Segate-ekb committed:
Инкрементальная загрузка на основании расхождений не только с предыдущим коммитом, но и с текущей конфигурацией (#637)

* Инкриментальная загрузка на основании расхождений не только с предыдущим коммитом, но и с текущей конфигурацией

* Фикс замечаний

* Первая имплементация

* Revert "Первая имплементация"

This reverts commit e3a2dfbd8d3b2af76a378f8a7593b3870d29559e.

* Притащил изменения из develop

* Принес изменения из develop

* Инкриментальная загрузка на основании расхождений не только с предыдущим коммитом, но и с текущей конфигурацией

Фикс замечаний

Первая имплементация

Revert "Первая имплементация"

This reverts commit e3a2dfbd8d3b2af76a378f8a7593b3870d29559e.

Принес изменения из develop

* Переехал на git-runner для более понятного управления выводом.
Добавил в тесты проверку загрузки незакоммиченых изменений

* Замечания сонара

* Убрал кириллицу... ну не любит ее CI

* Добавил тесты инкрементальной загрузки с изменениями в конфигураторе

* Вывод для CI

* Опечатка

* Установил Quotepath для работы инкрементальной загруки

* Исправил quotepath

* fix опечатка

* Опечатка в тесте

* Еще опечатка

* Добавил Нрег

* ТЕперь в ibcmd передается полный путь к файлу.

* Добавил тест, в котором есть проверка инкрементальной загрузки из файла ibcmd

* fix для возможности работы тестов без установленных параметров git

* Для систем без настроек

* Убрал лишнее

* Фиксы замечаний

* Фикс замечаний

* Добавил расшифровку к асерту

Segate-ekb committed:
Реализация использования ibcmd для серверных баз (#639)

* Первая имплементация

* Тест вывода

* Альфа вариант.

Изменено поведение менеджеров сборки для корректной работы ibcmd с серверными базами

* Добавил возможность устанавливать приоритетные настройки.

Теперь в файле настройки конкртеной команды - обладают большим приоритетом, чем дефолтные. Таким образом можно гибко рулить настройками для различных команд(Напрмер собирать с ibcmd, а выгружать без и тд.)

* Добавил в дефолтную схему ibcmd

* Первая имплементация

* Тест вывода

* Ребейз из девелопа

* Добавил возможность устанавливать приоритетные настройки.

Теперь в файле настройки конкртеной команды - обладают большим приоритетом, чем дефолтные. Таким образом можно гибко рулить настройками для различных команд(Напрмер собирать с ibcmd, а выгружать без и тд.)

* Добавил в дефолтную схему ibcmd

* Фикс тестов

* Рефакторинг

* вернул ошибочное форматирование

* Вынес изменения по чтению параметров в отдельный ПР

* fix замечаний

* Добавил приведение написания переданных ДБ к каноническому.

* фикс сонара

* Отработал замечания

yukon39 committed:
Merge branch 'issue-619' of https://github.com/yukon39/vanessa-runner into issue-619

yukon39 commented:
Открыт #645
